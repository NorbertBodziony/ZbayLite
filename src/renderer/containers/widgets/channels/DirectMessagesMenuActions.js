import React from 'react'

import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import * as R from 'ramda'
import { withRouter } from 'react-router-dom'

import ChannelMenuActionComponent from '../../../components/widgets/channels/ChannelMenuAction'
import { actionCreators } from '../../../store/handlers/modals'
import contactsHandler from '../../../store/handlers/contacts'
import dmChannelSelectors from '../../../store/selectors/directMessageChannel'
import { DateTime } from 'luxon'

export const mapStateToProps = state => ({
  targetAddress: dmChannelSelectors.targetRecipientAddress(state)
})

export const mapDispatchToProps = (dispatch, { history, directMessage, offer, ...rest }) => {
  return bindActionCreators(
    {
      onInfo: actionCreators.openModal('channelInfo'),
      onMute: () => console.warn('[ChannelMenuAction] onMute not implemented'),
      onDelete: target => contactsHandler.epics.deleteChannel({ ...target, history })
    },
    dispatch
  )
}
const ChannelMenuAction = ({ onDelete, targetAddress, ...props }) => {
  return (
    <ChannelMenuActionComponent
      onDelete={() =>
        onDelete({ address: targetAddress, timestamp: parseInt(DateTime.utc().toSeconds()) })
      }
      {...props}
    />
  )
}
export default R.compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(ChannelMenuAction)
