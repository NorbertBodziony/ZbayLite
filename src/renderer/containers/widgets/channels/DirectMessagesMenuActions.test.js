/* eslint import/first: 0 */
jest.mock('../../../vault')

import { mapDispatchToProps } from './DirectMessagesMenuActions'

describe('DirectMessagesMenuActions', () => {
  it('will receive right actions', async () => {
    const actions = mapDispatchToProps(x => x, { history: 'test' })
    expect(actions).toMatchSnapshot()
  })
})
