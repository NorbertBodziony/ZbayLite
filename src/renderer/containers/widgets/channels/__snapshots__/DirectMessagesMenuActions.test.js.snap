// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectMessagesMenuActions will receive right actions 1`] = `
Object {
  "onDelete": [Function],
  "onInfo": [Function],
  "onMute": [Function],
}
`;
